### PR TITLE
Migrate tests to qmake

### PR DESCRIPTION
**Pull request by orgads**

### CODE REVIEW

**Comment by orgads** [comment]
ping

**Comment by hugopl** [comment]
Sorry, I did read it but I was travelling until Sunday. I promise to review and merge both patches until next weekend.

File: ruby.pro [hunk 1]
```diff
@@ -21,6 +21,10 @@ SOURCES += RubyPlugin.cpp \
     projectmanager/RubyProjectManager.cpp \
     projectmanager/RubyProjectWizard.cpp
 
+equals(TEST, 1) {
+    SOURCES += tests/ScannerTest.cpp
```
Comment (hugopl) [inline ruby.pro]: this should be editor/ScannerTest.cpp

**Comment by hugopl** [comment]
Sorry for my ignorance about how QtC deal with tests, but after enabling the tests, how do I run them? Where is the executable?

I also disagree with the QtC standard of mix test code with the implementation code, but this is another topic.

Thanks.

**Comment by orgads** [comment]
The commit message mentions it:

> Execute them by running qtcreator -test Ruby

**Comment by hugopl** [comment]
oh hell... so I need the entire QtCreator compiled with the test flag enabled...

I have the bad, but practical, habit of compile and link the plugin against my system copy of QtCreator until I really need to debug something more deeply.

Thanks, no need to update the MR with the fix in the .pro file, I can update it here and push the changes, thanks!

**Comment by orgads** [comment]
> oh hell... so I need the entire QtCreator compiled with the test flag enabled...

Actually there *are* standalone tests in the Creator code-base. If your tests depend on 1-2 classes which can be compiled with the test, you can extract it. See for example tests/auto/valgrind.

> Thanks, no need to update the MR with the fix in the .pro file, I can update it here and push the changes, thanks!

Already did, sorry :)

**Comment by hugopl** [comment]
nice, thanks for th einfo... I'll compile qtc here, dinner, test then push :-)

**Pull request merged**